cdc/sorter/leveldb/system/system.go

```
@@ -0,0 +1,247 @@
+	cerrors "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/syndtr/goleveldb/leveldb"
+	"go.uber.org/zap"
+)
+
+// The interval of collecting leveldb metrics.
+const defaultMetricInterval = 15 * time.Second
+
+// State of a system.
+type sysState int
+
+const (
+	sysStateInit sysState = iota
+	sysStateStarted
+	sysStateStopped
+)
+
+// System manages leveldb sorter resource.
+type System struct {
+	dbs         []*leveldb.DB
+	dbSystem    *actor.System
+	dbRouter    *actor.Router
+	cleanSystem *actor.System
+	cleanRouter *actor.Router
+	cfg         *config.SorterConfig
+	closedCh    chan struct{}
+	closedWg    *sync.WaitGroup
+
+	state   sysState
+	stateMu *sync.Mutex
+}
+
+// NewSystem returns a system.
+func NewSystem(cfg *config.SorterConfig) *System {
+	dbSystem, dbRouter := actor.NewSystemBuilder("sorter").
+		WorkerNumber(cfg.LevelDB.Count).Build()
+	cleanSystem, cleanRouter := actor.NewSystemBuilder("cleaner").
+		WorkerNumber(cfg.LevelDB.Count).Build()
+	return &System{
+		dbSystem:    dbSystem,
+		dbRouter:    dbRouter,
+		cleanSystem: cleanSystem,
+		cleanRouter: cleanRouter,
+		cfg:         cfg,
+		closedCh:    make(chan struct{}),
+		closedWg:    new(sync.WaitGroup),
+		state:       sysStateInit,
+		stateMu:     new(sync.Mutex),
+	}
+}
+
+// ActorID returns an ActorID correspond with tableID.
+func (s *System) ActorID(tableID uint64) actor.ID {
+	h := fnv.New64()
+	b := [8]byte{}
+	binary.LittleEndian.PutUint64(b[:], tableID)
+	h.Write(b[:])
+	return actor.ID(h.Sum64() % uint64(s.cfg.LevelDB.Count))
+}
+
+// Router returns leveldb actors router.
+func (s *System) Router() *actor.Router {
+	return s.dbRouter
+}
+
+// CleanerRouter returns cleaner actors router.
+func (s *System) CleanerRouter() *actor.Router {
+	return s.cleanRouter
+}
+
+func (s *System) broadcast(ctx context.Context, router *actor.Router, msg message.Message) {
+	dbCount := s.cfg.LevelDB.Count
+	for id := 0; id < dbCount; id++ {
+		_ = router.SendB(ctx, actor.ID(id), msg)
```

why ignore the error here?